pyrealm/canopy_model/model/community_alt.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from numpy.typing import NDArray
+import numpy as np
+
+from pyrealm.canopy_model.model.flora import Flora
+from pyrealm.canopy_model.model.canopy import Canopy
+
+
+class Community:
+    def __init__(
+        self,
+        cell_id: int,
+        cell_area: float,
+        cohort_dbh: NDArray[np.float],
+        cohort_individuals: NDArray[np.int],
+        cohort_pfts: NDArray[np.str],
+        flora: Flora,
+    ):
+        self.cell_id = cell_id
+        self.cell_area = cell_area
+        self.cohort_dbh = cohort_dbh
+        self.cohort_individuals = cohort_individuals
+        self.cohort_pfts = cohort_pfts
+        self.flora = flora
+
+        # Arrays of required PFT traits for cohorts
+        self.traits = CommunityTraits(self.cohort_pfts, self.flora)
+
+        # Calculate geometries from initial dbh
+        self.tmodel_geometries = TModelGeometries(self.cohort_dbh, self.traits)
+
+    @classmethod
+    def from_json(
+        cls, path: Path, cell_id: int, cell_area: float, flora: Flora
+    ) -> Community:
+        """Factory method creating a Community from a JSON file."""
+
+        cohort_pfts, cohort_individuals, cohort_dbh = do_a_thing_with(path)
+
+        return Community(
+            cell_area=cell_area,
+            cell_id=cell_id,
+            cohort_dbh=cohort_dbh,
+            cohort_individuals=cohort_individuals,
+            cohort_pfts=cohort_pfts,
+            flora=flora,
+        )
+
+    def set_canopy(self) -> None:
+        """Or this could return a Canopy instance."""
+
+        self.canopy = Canopy(...)
+
+
+class CommunityTraits:
+    def __init__(self, cohort_pfts: NDArray[np.str], flora: Flora):
+        self.qm = self.get_trait_array("qm", cohort_pfts, flora)
+        ...
+
+    def get_trait_array(
+        self, trait_name: str, cohort_pfts: NDArray[np.str], flora: Flora
+    ) -> NDArray:
+        """Extract array of values across cohorts for a given trait.
```

Comment:
Actually - this is a static method isn't it. It could be a standalone function (easier to use in other use cases), or we could make `cohort_pfts` and `flora` into attributes and simplify the signature.